copy.go

```diff
@@ -185,7 +212,11 @@ func structToStruct(filter FieldFilter, src, dst *reflect.Value, userOptions *op
 		if !dst.CanSet() {
 			return errors.Errorf("dst %s, %s is not settable", dst, dst.Type())
 		}
-		dst.Set(*src)
+		if dst.Kind() == reflect.Ptr {
+			dst.Set(src.Addr())
```

Comment:
Is it possible that `src` is not addressable here? If so the corresponding check should be added.